account_journal_general_sequence/models/account_journal.py

```diff
@@ -13,27 +13,39 @@ class AccountJournal(models.Model):
     entry_number_sequence_id = fields.Many2one(
         comodel_name="ir.sequence",
         string="Account entry number sequence",
-        default=lambda self: self._default_entry_number_sequence(),
+        compute="_compute_entry_number_sequence",
+        domain="[('company_id', '=', company_id)]",
+        check_company=True,
+        readonly=False,
+        store=True,
         copy=False,
         help="Sequence used for account entry numbering.",
     )
 
-    def _default_entry_number_sequence(self):
+    @api.depends("company_id")
+    def _compute_entry_number_sequence(self):
         """Get the default sequence for all journals."""
-        result = self.env["ir.sequence"].search(
-            [("code", "=", "account_journal_general_sequence.default")]
-        )
-        if result:
-            return result
-        _logger.info("Creating default sequence for account move numbers")
-        result = self.env["ir.sequence"].create(
-            {
-                "name": _("Account entry default numbering"),
-                "code": "account_journal_general_sequence.default",
-                "implementation": "no_gap",
-                "prefix": "%(range_year)s/",
-                "padding": 10,
-                "use_date_range": True,
-            }
-        )
-        return result
+        for one in self:
+            sequence = self.env["ir.sequence"].search(
+                [
+                    ("code", "=", "account_journal_general_sequence.default"),
+                    ("company_id", "=", one.company_id.id),
+                ]
+            )
+            if not sequence:
+                _logger.info("Creating default sequence for account move numbers")
+                sequence = self.env["ir.sequence"].create(
+                    {
+                        "name": _(
+                            "Account entry default numbering (%s)",
+                            one.company_id.name,
+                        ),
+                        "code": "account_journal_general_sequence.default",
+                        "company_id": one.company_id.id,
+                        "implementation": "no_gap",
+                        "prefix": "%(range_year)s/",
+                        "padding": 6,
```

Comment:
Leave it like it was, so you don't have to change the preexisting test. This way you avoid introducing more breaking changes than needed.

```suggestion
                        "padding": 10,
```